runtime/reconcilers/model.go

```diff
@@ -1030,32 +1031,51 @@ func (r *ModelReconciler) executeAll(ctx context.Context, self *runtimev1.Resour
 	defer release()
 
 	// First step is to resolve and sync the partitions.
-	err = r.resolveAndSyncPartitions(ctx, self, model, incrementalState)
-	if err != nil {
-		return "", nil, false, fmt.Errorf("failed to sync partitions: %w", err)
+	// Skip this if we're doing an explicit partition refresh (TriggerPartitions), as we only want to execute the explicitly triggered partitions
+	if !model.Spec.TriggerPartitions {
```

Comment:
You can have a case where e.g. both `.TriggerFull` _and_ `.TriggerPartitions` are true (for example, the user may first refresh a partition, then before that operation is done, trigger a full refresh). So this check is not sufficient to determine whether to discover new partitions or not.